deepchem/data/data_loader.py

```diff
@@ -875,9 +876,71 @@ class provides convenience files to lead FASTA data and
   learning tasks.
   """
 
-  def __init__(self):
-    """Initialize loader."""
-    pass
+  def __init__(self,
+               featurizer: Optional[Featurizer] = None,
+               auto_add_annotations: bool = False,
+               legacy: bool = True):
+    """Initialize FASTALoader.
+
+    Parameters
+    ----------
+    featurizer: Featurizer (default: None)
+      The Featurizer to be used for the loaded FASTA data.
+
+      If featurizer is None and legacy is True, the original featurization
+      logic is used, creating a one hot encoding of all included FASTA strings
+      of shape
+      (number of FASTA sequences, number of channels + 1, sequence length, 1).
+
+      If featurizer is None and legacy is False, the featurizer is initialized
+      as a OneHotFeaturizer object with charset ("A", "C", "T", "G") and
+      max_length = None.
+
+    auto_add_annotations: bool (default False)
+      Whether create_dataset will automatically add [CLS] and [SEP] annotations
+      to the sequences it reads in order to assist tokenization.
+      Keep False if your FASTA file already includes [CLS] and [SEP] annotations.
+
+    legacy: bool (default True)
+      Whether to use legacy logic for featurization. Legacy mode will create
+      a one hot encoding of the FASTA content of shape
+      (number of FASTA sequences, number of channels + 1, max length, 1).
+
+      Legacy mode is only tested for ACTGN charsets, and will be deprecated.
+   """
+
+    # Process legacy toggle
+    if legacy:
+      logger.info("""
+                  Deprecation warning: Legacy mode will soon be deprecated.
+                  Disable legacy mode by passing legacy=False during
+                  construction of FASTALoader object.
+                  """)
+      if featurizer is not None or auto_add_annotations:
+        logger.warning(f"""
```

Comment:
I'd say just raise a value error here instead of trying to correct. Usually better to fail fast rather than proceed since users don't notice warnings often